### ISSUE TITLE
Ability to export report to CSV OR fix copy-paste of tables from report

### ISSUE THREAD

**Issue by Kristian-Popov** [moneymanagerex/moneymanagerex#4256]
# Description of the feature

I would really like to have ability to export some report (e.g. categories summary) to CSV file or some other format to process it in a spreadsheet.

If I understand correctly, there's ability to export to Excel (#3861), but I don't use that and I'd like to have a solution that works on Linux.

For example, for a report like this:
![Untitled](https://user-images.githubusercontent.com/19193068/158956176-b63fc829-d4dc-440f-af5e-c157ea94bd0d.png)

I would like to get the following example CSV file:
```"Kategoria","Kwota","Suma"
"Czynsz","5","5"
"Darowizny","10","10"
```

# Alternative proposal
I can copy-paste data from tables in report, but rows where actual numbers can be found are smashed together without any kind of separators. In case of pasting to LibreOffice Calc, entire row is smashed into one cell. This is example what I got for a report similar to found above:

```Kategoria
Kwota
Suma
Czynsz 5 5
Darowizny:Inne darowizny 10 10
```

My wild guess is adding some kind of separator character into text that is copied should be enough for Calc to split data into columns.

# Use cases

As a user of MMEx, I want to get ability to export/copy data from report into a spreadsheet for further analytics or processing.

**Comment by vomikan** [comment]
Could you try export the report to html or xml format then open it in LibreOffice Calc?
![image](https://user-images.githubusercontent.com/6836805/158976947-d50193be-c7c2-46f7-ba21-ef4b4a6771fd.png)

**Comment by tactilis** [comment]
@Kristian-Popov

I don't think this is an MMEX problem.

It may be a Linux-specific problem, because on Windows, the content of the report pastes does past correctly into Excel _and_ into LibreOffice Calc.

I've tried the same approach as I detailed here [https://github.com/moneymanagerex/moneymanagerex/issues/3861#issuecomment-920668895](https://github.com/moneymanagerex/moneymanagerex/issues/3861#issuecomment-920668895).

Here is an example, using the Categories Summary:

1. Hide the charts
2. Click Expand All

![image](https://user-images.githubusercontent.com/4060092/158970445-1e99bb45-ae91-4696-9395-8c8fd924a2d2.png)
 
 
3. Press CTRL+A in MMEX to select the entire report

![image](https://user-images.githubusercontent.com/4060092/158971742-d3cb92ac-b21a-440e-9ee7-239c81ef72c0.png)
 
 

4. Press CTRL+V to paste into a LibreOffice Calc
5. Press CTRL+A in the spreadsheet to select all cells
6. Double click on a column boundary to space all the columns properly
 
![image](https://user-images.githubusercontent.com/4060092/158974695-44ec65b2-5bef-49bd-ae9e-99eaf7b11975.png)

Unfortunately, Calc doesn't let you right click and 'Remove Hyperlinks' when you have multiple cells selected, so to get rid of the hyperlinks:

7. Click on the Column A header to select the entire column.
8. Press CTRL+C to copy the entire column
9. Right click on cell A1 and click Paste Special | Unformatted Text.
10. Click OK on the Text Import dialog that is displayed.

![image](https://user-images.githubusercontent.com/4060092/158978720-cc357f62-cac9-40a6-8c99-6010a80c0d57.png)

11. Now format the figures as Currency

![image](https://user-images.githubusercontent.com/4060092/158978905-8a0fc86e-0c08-4547-8914-a2f624872cac.png)
 
 
You could write a macro to automate all of those steps.

 
Does the above procedure work for you on Linux?  If it doesn't, then the problem is the way that Linux (X Window or Wayland) handles the clipboard.

**Comment by Kristian-Popov** [comment]
@vomikan actually your solution worked. I tried exporting to CSV from the same menu and got normal transaction export window, so I assumed that exporting report from menu "File -> Export" is not possible at all. I am able to copy table from exported HTML to Calc.

@tactilis I tried your suggestion, unfortunately it still did not work. I noticed that on my system space is used both as a separator and thousand digits separator, so this problem may be locale-dependent.
![image](https://user-images.githubusercontent.com/19193068/159157835-1cc78c53-301a-4539-8dd9-f45ea003fc04.png)
```
Czynsz 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 -11 100.00 -11 100.00 

Przychody 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 
Wydatki 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 -11 100.00 -11 100.00 
Suma 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 -11 100.00 -11 100.00 
```
About info of MMEx:
```
Money Manager Ex
Wersja: 1.5.13 64-bit
• Zbudowany lut 6 2022 13:11:14
• Wersja bazy danych: 7 • (aes128cbc)
• Migawka Git: d019150 (2022-02-06)
MMEX wykorzystuje następujące produkty:
• wxWidgets 3.1.5 (wxMSW 10.0)
• wxSQLite3 4.6.0 (SQLite 3.31.1)
• RapidJSON 1.1.0
• Lua 5.3.5
• lunasvg v2.0.1
• libcurl/7.61.0-DEV WinSSL
• gettext 0.19.8.1
• apexcharts.js
Zbuduj za pomocą:
• CMake 3.16.2
• MSBuild 15.9.21.664
• Microsoft Visual Studio 15.0
• Microsoft Visual C++ 19.16.27035.0
• Windows SDK 10.0.14393.0
Działający na:
• Windows 10 (build 19043), 64-bit edition
• English (U.S.) (windows-1252)
• 1920x1080 32bit 96x96ppi
```

In conclusion, I am able to do what I planned, feel free to close the issue.
